### PR TITLE
Fix the YARN resource leakage issue caused by the HikariCP connection…

### LINKED ISSUE #6092: Hive on Spark problem
**For better global communication, Please describe it in English. If you feel the description in English is not clear, then you can append description in Chinese(just for Mandarin(CN)), thx! **
**Describe the bug**

**To Reproduce**
Steps to reproduce the behavior, for example:

**Expected behavior**

**Screenshots**

**Which version of Dolphin Scheduler:**
 -[1.1.0-preview]

**Additional context**

**Requirement or improvement**
- Please describe about your requirements or improvement suggestions.

### PR DESCRIPTION
**Pull request by QQhuxuhui**
Fix the YARN resource leakage issue caused by the HikariCP connection pool

fix: [#6092](https://github.com/apache/dolphinscheduler/issues/6092)

## Purpose of the pull request

The purpose of this submission is to address the resource leakage issue caused by the connection pool. Starting from version 2.x, third-party data sources utilize the HikariCP connection pool. Here is a description of the problem:

1. When running tasks in Hive on Spark mode, the JDBC connection in Hive is heavier compared to regular RDBMS connections. This is especially true in Hive on Spark mode because when the underlying connection needs to execute SQL, HS2 (HiveServer2) requests CONTAINER resources from YARN and starts a distributed Spark on YARN cluster to execute the compiled SQL in a distributed manner. After the SQL execution is complete, the Spark on YARN resources are not immediately released. Instead, they are kept for a certain period of time to reuse these resources for executing new SQL submitted through the same JDBC connection by the client. The Spark on YARN resources are only completely released when the JDBC connection is closed or when the configured timeout is reached without any new SQL submissions from the client.
2. When using a database connection pool technology, closing a JDBC connection essentially returns the connection to the connection pool without actually closing the underlying JDBC connection. As a result, the Spark on YARN resources associated with the connection are not released promptly, leading to resource leakage. Consequently, other jobs that request resources from YARN have to wait in a queue, affecting the execution of those jobs.
3. In the project, HikariCP is used as the database connection pool, and the idle timeout duration (idletimeout) for the database connections is not configured. The effective idle timeout duration is the default value of 10 minutes configured in the HikariCP source code. Therefore, after each connection's underlying SQL job is completed, it takes 10 minutes for the associated Spark on YARN resources to be truly released. This results in other jobs queuing and waiting for YARN resources.

The approach in this solution is to adjust the parameters of the database connection pool, especially the minimum connection count and idle timeout time, in order to close idle database connections more quickly and actively. For example, configuring the IdleTimeout to 30 seconds and setting the MinimumIdle to 0. This means that all connections will be closed 30 seconds after the SQL job finishes, releasing all the Spark on YARN resources and resolving the resource leakage issue.

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/provider/JDBCDataSourceProvider.java [hunk 1]
```diff
@@ -56,9 +56,10 @@ public static HikariDataSource createOneSessionJdbcDataSource(BaseConnectionPara
 
         Boolean isOneSession = PropertyUtils.getBoolean(Constants.SUPPORT_HIVE_ONE_SESSION, false);
         dataSource.setMinimumIdle(
-                isOneSession ? 1 : PropertyUtils.getInt(DataSourceConstants.SPRING_DATASOURCE_MIN_IDLE, 5));
+                isOneSession ? 1 : PropertyUtils.getInt(DataSourceConstants.SPRING_DATASOURCE_MIN_IDLE, 0));
         dataSource.setMaximumPoolSize(
                 isOneSession ? 1 : PropertyUtils.getInt(DataSourceConstants.SPRING_DATASOURCE_MAX_ACTIVE, 50));
+        dataSource.setIdleTimeout(PropertyUtils.getInt(DataSourceConstants.SPRING_DATASOURCE_IDLE_TIMEOUT, 30000));
```
Comment (ruanwenjun) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-api/src/main/java/org/apache/dolphinscheduler/plugin/datasource/api/provider/JDBCDataSourceProvider.java:62]: Using the Hikari's default value is better.

File: dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/constants/DataSourceConstants.java [hunk 2]
```diff
@@ -120,6 +120,8 @@ public class DataSourceConstants {
 
     public static final String SPRING_DATASOURCE_TEST_ON_BORROW = "spring.datasource.testOnBorrow";
 
+    public static final String SPRING_DATASOURCE_IDLE_TIMEOUT = "spring.datasource.idleTimeoutMs";
```
Comment (ruanwenjun) [inline dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/constants/DataSourceConstants.java:123]: Please add doc about how to using this configuration.

**Review by ruanwenjun — COMMENTED** [review]
LGTM, but the connection pool has been removed by #14631, so the connection leak should already been solved.

**Comment by QQhuxuhui** [comment]
[14631](https://github.com/apache/dolphinscheduler/pull/14631) solved this problem

**Pull request closed without merging**